### PR TITLE
Replace Utf16Utility.GetPointerToFirstInvalidChar fallback with Vector128

### PR DESCRIPTION
**Pull request by stephentoub**
The else block will only be used on platforms where `Vector<T>` is 128-bit, so just use Vector128 explicitly.

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf16Utility.Validation.cs [hunk 1]
```diff
@@ -340,16 +340,16 @@ internal static unsafe partial class Utf16Utility
...
                         // handle that shortly.
 
-                        Vector<ushort> utf16Data = Unsafe.ReadUnaligned<Vector<ushort>>(pInputBuffer);
-                        Vector<ushort> twoOrMoreUtf8Bytes = Vector.GreaterThanOrEqual(utf16Data, vector0080);
-                        Vector<ushort> threeOrMoreUtf8Bytes = Vector.GreaterThanOrEqual(utf16Data, vector0800);
-                        Vector<nuint> sumVector = (Vector<nuint>)(Vector<ushort>.Zero - twoOrMoreUtf8Bytes - threeOrMoreUtf8Bytes);
+                        Vector128<ushort> utf16Data = Vector128.Load((ushort*)pInputBuffer);
+                        Vector128<ushort> twoOrMoreUtf8Bytes = Vector128.GreaterThanOrEqual(utf16Data, vector0080);
+                        Vector128<ushort> threeOrMoreUtf8Bytes = Vector128.GreaterThanOrEqual(utf16Data, vector0800);
+                        Vector128<nuint> sumVector = Vector128.As<ushort, nuint>(Vector128<ushort>.Zero - twoOrMoreUtf8Bytes - threeOrMoreUtf8Bytes);
```
Comment (EgorBo) [inline src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf16Utility.Validation.cs:346]: nit: `.AsNUint()` but presumably this whole path should be eventually removed once we improve the top one for ARM64

**Review by EgorBo — APPROVED** [review]

**Comment by xtqqczze** [comment]
Is `Vector128` now hardware accelerated on wasm?

**Comment by stephentoub** [comment]
> Is `Vector128` now hardware accelerated on wasm?

Yes

**Comment by xtqqczze** [comment]
@stephentoub Could you also take a look at `Ascii.Utility.cs`?

https://github.com/dotnet/runtime/blob/7bc9f6bd9652fcfcb850d7ac3f52f9a1bb9d1431/src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Utility.cs#L1057

**Comment by stephentoub** [comment]
> @stephentoub Could you also take a look at `Ascii.Utility.cs`?
> 
> https://github.com/dotnet/runtime/blob/7bc9f6bd9652fcfcb850d7ac3f52f9a1bb9d1431/src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Utility.cs#L1057

I already did.
https://github.com/dotnet/runtime/pull/90390

**Review by tannergooding — APPROVED** [review]

**Comment by stephentoub** [comment]
Failure is known https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=371143

**Pull request merged**